generate/main.go

```diff
@@ -208,6 +208,7 @@ func main() {
 				{name: "savepoint_stmt", inline: []string{"savepoint_name"}},
 				{name: "select_stmt", inline: []string{"select_no_parens", "simple_select", "opt_sort_clause", "select_limit"}},
 				{name: "set_stmt", inline: []string{"set_rest", "set_rest_more", "generic_set"}, exclude: regexp.MustCompile("CHARACTERISTICS"), replace: map[string]string{"'TRANSACTION' transaction_mode_list | ": ""}},
+				{name: "set_database", stmt: "set_stmt", inline: []string{"set_rest", "set_rest_more", "generic_set"}, exclude: regexp.MustCompile("CHARACTERISTICS"), replace: map[string]string{"var_name": "'DATABASE'", "'DEFAULT'": "var_list", "var_list": "database_name", "'LOCAL'": "", "'SESSION'": "", "'TRANSACTION' transaction_mode_list | ": ""," | 'TIME' 'ZONE' zone_value ": ""},unlink: []string{"database_name"}},
```

Comment:
@mjibson, we figured it out using more replacements, but I forgot to update the link. You can see it here: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/dfafce3078cf944c7eb72fe42f9125f24e8691c4/set-database.html

I believe I tried with `match` and couldn't get it to work. If you have a sec to help out, I'm happy to try again, if you like.